apps/internal/base/internal/storage/storage.go

```diff
@@ -121,7 +129,7 @@ func (m *Manager) Read(ctx context.Context, authParameters authority.AuthParams)
 	if err != nil {
 		return TokenResponse{}, err
 	}
-	account, err := m.readAccount(homeAccountID, metadata.Aliases, realm)
+	account, err = m.readAccount(homeAccountID, metadata.Aliases, realm)
```

Comment:
Will the value of `account` be different after this call?

Reply:
If it finds an account, then it will and should be the same as the input account.
If an err is returned, this would be a blank `shared.Account{}` object

Reply:
I see.  Since we're now accepting `Account` as a parameter do we even need to call `m.readAccount()`?  Seems redundant now.

Reply:
I think you are right but the readAccount() function performs a few more checks to verify it is the correct account exists. I see there is a homeaccount id check which should always be true, because that is extracted from account in the previous step. However there is a realm check and environment aliases check which is from the initial configuration. 

I can dig a little more later to check if this is really redundant. Does that work?

Reply:
Yep!

Reply:
Tracking [here ](https://github.com/AzureAD/microsoft-authentication-library-for-go/issues/179)